src/execution/volcano/mod.rs

```diff
@@ -75,7 +77,12 @@ pub fn build_read<T: Transaction>(plan: LogicalPlan, transaction: &T) -> BoxedEx
             let right_input = childrens.pop().unwrap();
             let left_input = childrens.pop().unwrap();
 
-            HashJoin::from((op, left_input, right_input)).execute(transaction)
+            match &op.on {
+                JoinCondition::On { on, .. } if !on.is_empty() => {
```

Comment:
I'm very sorry. I checked it carefully again on my computer and found that there are three slt files that need to open the comment case.
- `tests\slt\sql_2016\F041_08.slt`
- `sqlite.slt` L33
- `subquery.slt` L37

At the same time, I discovered a new bug: still using `HashJoin` when or exists in the Join condition leads to incorrect results.
case on `sqlite.slt` L33

Tips: I think you can add a tag to the `JoinOperator` to decide whether to use `NestLoopJoin` or `HashJoin` in the Bind stage.